src/core/evaluator.js

```diff
@@ -2385,6 +2385,12 @@ class PartialEvaluator {
         });
     }
 
+    function applyInverseRotation(x, y, transf) {
+      const scale = Math.hypot(transf[0], transf[1]);
+      const matrix = transf.slice(0, 4).map(c => c / scale);
+      return [matrix[0] * x + matrix[1] * y, matrix[2] * x + matrix[3] * y];
```

Comment:
It seems like this is adding some unnecessary parsing/allocations, since you first create a temporary Array and then iterate through it.
Could you not instead just do e.g. the follow, which ought to be a tiny bit more efficient from both a CPU/memory perspective?
```javascript
return [
  (transf[0] / scale) * x + (transf[1] / scale) * y,
  (transf[2] / scale) * x + (transf[3] / scale) * y
];
```